pkg/generators/models/testdata/cases/allof_self_reference/api.yaml

```diff
@@ -0,0 +1,17 @@
+openapi: 3.0.0
+info:
+  version: 0.1.0
+  title: Test
+
+components:
+  schemas:
+    ListItem:
+      type: object
+      properties:
+        value:
+          type: string
+        next:
+          allOf:
+            - $ref: "#/components/schemas/ListItem"
+            - description: the next item
+            - nullable: true
```

Comment:
Can we have a test when the reference is not nullable? I don't really understand how it would work with Go type system.

Reply:
It would just not have a pointer, but I can add this.

Reply:
If you write it without a pointer its not going to compile.

Reply:
well, this means we should validate it with the generator and don't accept a spec that has a non-nullable recursive reference.

Reply:
I don't think we can do so without actually compiling the resulting package in a final step. These non-nullable recursive types don't have to be direct. A could contain B, B contains C and C finally contains A again. We would have the same problem, but hidden in a way that you can't really spot it when parsing and constructing model A. We could possibly write a check for this that runs before every model generation and tries to traverse the model specs checking especially for this. 

This could be a bigger task to get it right, should I include it here?

Reply:
You're right, didn't think of a 2 step loop. Then let's merge this as it is.